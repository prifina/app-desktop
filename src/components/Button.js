import React from 'react';
import PropTypes from 'prop-types';
import styled from "styled-components";
import '../assets/scss/style.css';



/**
 * Primary UI component for user interaction
 */
const Button = ({ btnType = 'default', backgroundColor,variant, fullWidth, size,className, label, ...props }) => {
  // console.log('btnType',btnType)

  return (
    <button
      type="button"
      className={[ `storybook-button--${btnType}` ,(size) ? `storybook-button--${size}` : ``,(variant) ? `storybook-button--${variant}` : ``, (fullWidth) ? `storybook-button--btn-block` : '',(className)? className:``].join(' ')}
      {...props}
    >
      {label}
    </button>
  );
};

Button.propTypes = {
  /**
   * Is this the principal call to action on the page?
   */
  btnType:  PropTypes.oneOf(['default', 'primary', 'secondary']),
  /**
   * What background color to use
   */
  backgroundColor: PropTypes.string,
  /**
   * How large should the button be?
   */
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  /**
   * How variant should the button be?
   */
  variant: PropTypes.oneOf(['solid', 'outline']),
  /**
   * How fullWidth should the button be?
   */
  fullWidth: PropTypes.bool,
  /**
   * Button contents
   */
  label: PropTypes.string.isRequired,
  /**
   * Optional click handler
   */
  onClick: PropTypes.func,
  /**
   * Button contents
   */
  className: PropTypes.string.isRequired,


};

Button.defaultProps = {
  backgroundColor: null,
  btnType: 'default',
  size: 'medium',
  variant:'solid',
  fullWidth:'false',
  onClick: undefined,
  className:''
};

export default Button;