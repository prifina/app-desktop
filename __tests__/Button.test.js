import React, {createRef} from "react";
import renderer from "react-test-renderer";
import Button from "../src/components/Button";
import { ThemeProvider } from "@blend-ui/core";

test("Button renders correctly", () => {
    const tree = renderer.create(<ThemeProvider>
        <Button>Submit</Button>
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
})