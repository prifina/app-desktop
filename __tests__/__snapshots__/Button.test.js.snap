// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 fESRGN"
>
  <div
    className="sc-bdfBwQ gIDPWr"
  >
    <button
      className="storybook-button--default storybook-button--medium storybook-button--solid storybook-button--btn-block "
      type="button"
    />
  </div>
</div>
`;
